### ISSUE TITLE
[Feature] extract text from image

### ISSUE THREAD

**Issue by solonovamax** [FossifyOrg/Gallery#294]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Gallery/issues) or [closed](https://github.com/FossifyOrg/Gallery/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Gallery/discussions) or [closed](https://github.com/FossifyOrg/Gallery/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Gallery/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

Support detecting & extracting text from an image

### Why do you want this feature?

My built-in screenshot app has a feature that does this, which I find myself using decently often. *especially* when adding alt-text for posts on [fedi](https://fedi.tips/what-is-mastodon-what-is-the-fediverse/)

### Additional information

Here is an example from my built-in screenshot app:

https://github.com/user-attachments/assets/aac2cbe0-7116-4a03-b06e-b8522a779914

Note: this is *not* using some hack, such as the android accessibility api, but instead truly extracting text from the image. You'll notice that it extracts the text from the "H", "B", and "I" icons in the video. Although not shown in the video, it can also extract the text from a normal image.

**Comment by chesio** [comment]
I know the app has no longer "Simple" in its name, but I personally think this is out of scope for a "simple" gallery app.

**Comment by solonovamax** [comment]
> I know the app has no longer "Simple" in its name, but I personally think this is out of scope for a "simple" gallery app.

Hmmm, that is quite fair.

but at least with the issue here, if it's decided to be out of scope, then at least the issue exists for others who search for it.

**Comment by naveensingh** [comment]
> I personally think this is out of scope for a "simple" gallery app.

I agree. There's also the problem of app size increase due to ML models. Google's MLKit based text recognition models average around 4 MB per script.

**Comment by Aworldc** [comment]
Now I know this ain't happening, but I would just like to say that this would then unlock the possiblilty of searching images by text.

Maybe that's better suited for the file manager app, maybe because of the file size concerns above, there should be a seperate search app, but yeah.

**Comment by solonovamax** [comment]
ngl searching images by text is even less likely to happen than text extraction from images lol
